File: android/app/src/main/kotlin/com/wagmilabs/sigil/ui/implementorinit/MnemonicDisplayView.kt
package com.wagmilabs.sigil.ui.implementorinit

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.itemsIndexed
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ContentCopy
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

@Composable
fun MnemonicDisplayView(
    words: List<String>,
    confirmedWrittenDown: Boolean,
    onConfirmChange: (Boolean) -> Unit,
    onContinue: () -> Unit,
    onCopy: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .verticalScroll(rememberScrollState())
            .padding(24.dp),
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {
        Row(
            modifier = Modifier
                .background(
                    MaterialTheme.colorScheme.errorContainer,
                    RoundedCornerShape(8.dp)
                )
                .padding(16.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Icon(
                Icons.Default.Warning,
                contentDescription = null,
                tint = MaterialTheme.colorScheme.error
            )
            Text(
                "Write this down — you cannot recover it later",
                style = MaterialTheme.typography.bodyMedium,
                fontWeight = FontWeight.SemiBold,
                color = MaterialTheme.colorScheme.onErrorContainer
            )
        }

        LazyVerticalGrid(
            columns = GridCells.Fixed(3),
            verticalArrangement = Arrangement.spacedBy(8.dp),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            modifier = Modifier.height(400.dp)
        ) {
            itemsIndexed(words) { index, word ->
                Row(
                    modifier = Modifier
                        .background(
                            MaterialTheme.colorScheme.surfaceVariant,
                            RoundedCornerShape(4.dp)
                        )
                        .padding(8.dp)
                ) {
                    Text(
                        "${index + 1}.",
                        style = MaterialTheme.typography.bodySmall,
                        modifier = Modifier.width(24.dp)
                    )
                    Text(
                        word,
                        style = MaterialTheme.typography.bodyMedium,
                        fontWeight = FontWeight.Medium
                    )
                }
            }
        }

        OutlinedButton(onClick = onCopy) {
            Icon(Icons.Default.ContentCopy, contentDescription = null)
            Spacer(modifier = Modifier.width(8.dp))
            Text("Copy to Clipboard")
        }

        Row(verticalAlignment = Alignment.CenterVertically) {
            Checkbox(checked = confirmedWrittenDown, onCheckedChange = onConfirmChange)
            Text("I have written down all 24 words")
        }

        Button(
            onClick = onContinue,
            enabled = confirmedWrittenDown,
            modifier = Modifier.fillMaxWidth()
        ) {
            Text("Continue")
        }
    }
}

File: ios/Sources/UI/ImplementorInit/MnemonicDisplayView.swift
import SwiftUI

public struct MnemonicDisplayView: View {
    let words: [String]
    @Binding var confirmedWrittenDown: Bool
    let onContinue: () -> Void
    let onCopy: () -> Void

    public init(
        words: [String],
        confirmedWrittenDown: Binding<Bool>,
        onContinue: @escaping () -> Void,
        onCopy: @escaping () -> Void
    ) {
        self.words = words
        self._confirmedWrittenDown = confirmedWrittenDown
        self.onContinue = onContinue
        self.onCopy = onCopy
    }

    public var body: some View {
        ScrollView {
            VStack(spacing: 24) {
                HStack {
                    Image(systemName: "exclamationmark.triangle.fill")
                        .foregroundColor(.orange)
                    Text("Write this down — you cannot recover it later")
                        .fontWeight(.semibold)
                        .foregroundColor(.orange)
                }
                .padding()
                .background(Color.orange.opacity(0.1))
                .cornerRadius(8)

                LazyVGrid(columns: Array(repeating: GridItem(.flexible()), count: 3), spacing: 12) {
                    ForEach(Array(words.enumerated()), id: \.offset) { index, word in
                        HStack {
                            Text("\(index + 1).")
                                .foregroundColor(.secondary)
                                .font(.caption)
                                .frame(width: 24, alignment: .trailing)
                            Text(word)
                                .font(.body)
                                .fontWeight(.medium)
                                .frame(maxWidth: .infinity, alignment: .leading)
                        }
                        .padding(8)
                        .background(Color(.systemGray6))
                        .cornerRadius(4)
                    }
                }

                Button(action: onCopy) {
                    HStack {
                        Image(systemName: "doc.on.doc")
                        Text("Copy to Clipboard")
                    }
                }
                .buttonStyle(.bordered)

                Toggle(isOn: $confirmedWrittenDown) {
                    Text("I have written down all 24 words")
                        .font(.body)
                }

                Button("Continue") {
                    onContinue()
                }
                .buttonStyle(.borderedProminent)
                .disabled(!confirmedWrittenDown)
            }
            .padding()
        }
    }
}
